file-server/src/main/java/edu/lysak/fileserver/server/MyServerSocket.java

```diff
@@ -1,85 +1,46 @@
         this.address = address;
         this.port = port;
-        this.fileStorage = fileStorage;
+        this.responseHandler = responseHandler;
     }
 
     public void start() throws IOException {
-        System.out.println("Server started!");
-        while (true) {
-            try (ServerSocket server = new ServerSocket(port, 50, InetAddress.getByName(address))) {
-                try (
-                        Socket socket = server.accept();
-                        DataInputStream input = new DataInputStream(socket.getInputStream());
-                        DataOutputStream output = new DataOutputStream(socket.getOutputStream())
-                ) {
-                    String request = input.readUTF();
-//                    String request = new String(input.readAllBytes(), StandardCharsets.UTF_8);
 
-                    String response;
-                    if (request.startsWith(PUT)) {
-                        response = processPutRequest(request);
-                    } else if (request.startsWith(GET)) {
-                        response = processGetRequest(request);
-                    } else if (request.startsWith(DELETE)) {
-                        response = processDeleteRequest(request);
-                    } else if ("exit".equals(request)) {
-                        break;
-                    } else {
-                        throw new IllegalArgumentException(request + " request is not supported");
-                    }
+        try (ServerSocket server = new ServerSocket(port, 50, InetAddress.getByName(address))) {
+            System.out.println("Server started!");
 
-//                    output.writeUTF(response);
-                    output.write(response.getBytes(StandardCharsets.UTF_8));
+            ExecutorService executor = Executors.newFixedThreadPool(4);
+            while (!Thread.interrupted()) {
+                Socket socket = acceptClientSocket(server);
+                if (socket == null) {
+                    executor.shutdownNow();
+                    break;
                 }
-            }
-        }
-    }
 
-    private String processPutRequest(String request) throws IOException {
-        String fileNameAndData = request.substring(PUT.length() + 1);
-        String fileName = fileNameAndData.substring(0, fileNameAndData.indexOf(SPACE));
-        String fileContent = fileNameAndData.substring(fileNameAndData.indexOf(SPACE) + 1);
-        if (fileStorage.contains(fileName)) {
-            return "403";
-        }
-        fileStorage.addFile(fileName, fileContent);
-        return "200";
-    }
-
-    private String processGetRequest(String request) throws IOException {
-        String fileName = request.substring(GET.length() + 1);
-        if (fileStorage.contains(fileName)) {
-            return "200 " + fileStorage.getFileContent(fileName);
+                executor.submit(() -> new ClientHandler(responseHandler, server, socket).run());
```

Comment:
```suggestion
                executor.submit(new ClientHandler(responseHandler, server, socket));
```
You don't need to wrap it in lambda, `ClientHandler` implements `Runnable`.

Reply:
done